### PR TITLE
feat(revocation): add onchain claim revocation code

### PR DESCRIPTION
**Pull request by nichonien**
This PR adds code for revocation of onchain claims / roles. The code haas been added to the `onchain-claims`  package.

### CODE REVIEW

**Review by JGiter — DISMISSED** [review]

File: packages/onchain-claims/src/claim-revocation.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      const tx = await this._revocationRegistry.revokeClaim(
+        namespaceHash,
+        subjectAddress,
+        revokerAddress
+      );
+      const receipt = await tx.wait();
+      const event = receipt.events?.find((e: Event) => e.event === 'Revoked');
+      if (!event) return false;
+    } catch (error) {
+      if (error instanceof Error) {
```
Comment (Harasz) [inline packages/onchain-claims/src/claim-revocation.ts:49]: What was the reason to add such check?

File: packages/onchain-claims/src/claim-revocation.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+   * @returns Promise<boolean>
+   */
+  async revokeClaimforDIDs(
+    namespace: string,
+    subjects: string[],
+    revoker: string
+  ): Promise<boolean> {
+    const namespaceHash = utils.namehash(namespace);
+    const revokerAddress = addressOf(revoker);
+    const revocationSubjects = [];
```
Comment (Harasz) [inline packages/onchain-claims/src/claim-revocation.ts:72]:
```suggestion
    const revocationSubjects = subjects.map(subject => addressOf(subject))
```

**Review by Harasz — COMMENTED** [review]
I see a lot of eslint warnings and errors

**Review by JGiter — APPROVED** [review]

File: packages/onchain-claims/src/claim-revocation.ts (hunk 1 above)
Comment (nichonien) [reply packages/onchain-claims/src/claim-revocation.ts:49]: If you are asking about `if (error instanceof Error) `, then it was added when we upgraded `typescript`. For some reason the error regarding the use of `any` was not getting eliminated and it is there for all such call in `ew-did-registry` if I remember correctly.

Comment (nichonien) [reply packages/onchain-claims/src/claim-revocation.ts:49]: Most of the eslint warnings are for `no-unresolved`. I would put some time and work on it later.

File: packages/onchain-claims/src/claim-revocation.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
+import { Event, utils, providers } from 'ethers';
```
Comment (Harasz) [inline packages/onchain-claims/src/claim-revocation.ts:1]: `[@typescript-eslint/no-unused-vars] 'providers' is defined but never used.`

File: packages/onchain-claims/src/claim-revocation.ts (hunk 1 above)
Comment (Harasz) [reply packages/onchain-claims/src/claim-revocation.ts:49]: Okay, why we catch error and then re-throw it?

**Pull request merged**